### PR TITLE
Adds GiantSwarm to solutions page

### PR DESCRIPTION
**Pull request by rsokolkov**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.7 Features: set Milestone to `1.7` and Base Branch to `release-1.7`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box below to allow 
> reviewers fix problems on your patch and speed up the review process.
> Please delete this note before submitting the pull request.

This pull request adds [Giantswarm](https://giantswarm.io) to pick-right-solution page in documentantion.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3843)

### CODE REVIEW

File: docs/setup/pick-right-solution.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ a Kubernetes cluster from scratch.
 
 * [IBM Bluemix Container Service](https://console.ng.bluemix.net/docs/containers/container_index.html) offers managed Kubernetes clusters. Currently in beta.
 
+* [GiantSwarm.io](https://giantswarm.io/product/) offers managed Kubernetes clusters in own datacenter, on-premises or in cloud.
```
Comment (puja108) [inline docs/setup/pick-right-solution.md]: `in own datacenter, on-premises or in cloud.`-> `in their own datacenter, on-premises, or on public clouds.`

File: docs/setup/pick-right-solution.md [hunk 2]
```diff
@@ -150,6 +152,7 @@ KCluster.io          |              | multi-support | multi-support | [docs](htt
 KUBE2GO.io          |              | multi-support | multi-support | [docs](https://kube2go.io) | Commercial
 Madcore.Ai           | Jenkins DSL  | Ubuntu | flannel     | [docs](https://madcore.ai)                        | Community ([@madcore-ai](https://github.com/madcore-ai))
 Platform9        |              | multi-support | multi-support | [docs](https://platform9.com/managed-kubernetes/) | Commercial
+GiantSwarm.io        |              | CoreOS | flannel     | [docs](https://docs.giantswarm.io/) | Commercial
```
Comment (puja108) [inline docs/setup/pick-right-solution.md]: `flannel` -> `flannel and/or Calico`
(we use flannel only onprem and always together with Calico)

File: docs/setup/pick-right-solution.md (hunk 1 above)
Comment (marians) [inline docs/setup/pick-right-solution.md]: Please change `GiantSwarm.io` to `Giant Swarm`

File: docs/setup/pick-right-solution.md (hunk 2 above)
Comment (marians) [inline docs/setup/pick-right-solution.md]: Please change `GiantSwarm.io` to `Giant Swarm`

**Comment by rsokolkov** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/setup/pick-right-solution.md, line 63 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843#-KkUoOaVsENf2rlGvu3t-r2-63:-KkUtlaCZ5nF4nI4qviC:bb6un1j) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/5e5a6e1897f01089fe6f4de6250c2e04509c0407/docs/setup/pick-right-solution.md#L63)):*
<details><summary><i>Previously, marians (Marian Steinbach) wrote…</i></summary><blockquote>

Please change `GiantSwarm.io` to `Giant Swarm`
</blockquote></details>

Done

---

*[docs/setup/pick-right-solution.md, line 155 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843#-KkUoOaVsENf2rlGvu3t-r2-155:-KkUtu0fShnoSpc-Fm-0:bb6un1j) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/5e5a6e1897f01089fe6f4de6250c2e04509c0407/docs/setup/pick-right-solution.md#L155)):*
<details><summary><i>Previously, marians (Marian Steinbach) wrote…</i></summary><blockquote>

Please change `GiantSwarm.io` to `Giant Swarm`
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843)*

**Comment by rsokolkov** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/setup/pick-right-solution.md, line 155 at r2](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843#-KkUoOaVsENf2rlGvu3t-r2-155:-KkUvUTzw5-BEK8Dsxfe:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/5e5a6e1897f01089fe6f4de6250c2e04509c0407/docs/setup/pick-right-solution.md#L155)):*
<details><summary><i>Previously, rsokolkov (Roman S) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843)*

**Comment by puja108** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843#-:-KkWMUiXmVTVC72hq3pT:bnfp4nl)*

**Comment by puja108** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843)*

**Comment by marians** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes.github.io/3843#-:-KkWSYYWdSKTEBd1cBmu:bnfp4nl)*

**Pull request merged**